Python/iSense.py

```diff
@@ -0,0 +1,310 @@
+
+    def getCreationTime(self):
+        return self.__created_at
+
+    def ownerName(self):
+        return self.__owner
+
+    def ownerUrl(self):
+        return self.__owner_url
+
+    def getDataSetCount(self):
+        return self.__data_set_count
+
+    def getDataSetIds(self):
+        return self.__data_set_ids
+
+    def getFieldCount(self):
+        return self.__field_count
+
+    def getDataByField(self, field):
+        data = []
+        for data_set in self.__data_sets:
+            data.extend(data_set.getDataByField(field))
+
+        return data
+
+
+
+class DataSet(object):
+    # fields is an array. ex. [time, x, y]
+    # data is an array of arrays (each inner array represents the data for a field)
+
+    def __init__(self, dictionary):
+        self.__id = dictionary['id']
+        self.__name = dictionary['name']
+        self.__url = dictionary['url']
+        self.__path = dictionary['path']
+        self.__created_at = dictionary['createdAt']
+        self.__field_count = dictionary['fieldCount']
+        self.__data_point_count = dictionary['datapointCount']
+        self.__display_url = dictionary['displayURL']
+        self.__data = dictionary['data']
+
+
+    # Refresh pulls down the data sets latest data from isense
+    def refresh(self):
+        self.__init__(self.data)
+        return
+
+    def getId(self):
+        return self.__id
+
+    def getName(self):
+        return self.__name
+
+    def getPath(self):
+        return self.__path
+
+    def getCreationTime(self):
+        return self.__created_at
+
+    def getFieldCount(self):
+        return self.__field_count
+
+    def getDataPointCound(self):
+        return self.__data_point_count
+
+    def getDisplayUrl(self):
+        return self.__display_url
+
+    def getDataByField(self, field):
+        # return all data related to a field
+        data = []
+
+        for point in self.__data:
+            data.append(point[str(field.getId())])
+
+        return data
+
+    def getData(self):
+        return self.__data
+
+    # given another data object it will append it to this data set
+    def appendData(self, data_set_id, data_dictionary, credentials):
+
+        # Append the new data to the data set on iSENSE
+        if credentials.isUser():
+            response = iSenseAPI().appendDataSet(credentials.getUsername(), credentials.getPassword(),
+                data_set_id, data_dictionary)
+        elif credentials.isKey():
+            response = iSenseAPI().appendDataSet(credentials.contrib_key(),
+                data_set_id, data_dictionary)
+        else:
+            response = INVALID_CREDENTIALS 
+
+        # After the data is uploaded refresh this data set object so it has the latest data 
+        refresh()
+
+        # Response is a dictionary. It can be used to check if the post request was successful.
+        # It also will contain any errors the server may have returned.
+        return response
+
```

Comment:
The API also has an editData method.  Given how appendData is implemented, it should be trivial to adapt that to editing the entire data set.